core/integration/container_test.go

```diff
@@ -3982,3 +3983,22 @@ func TestContainerImageLoadCompatibility(t *testing.T) {
 		}
 	}
 }
+
+func TestContainerWithPlainSecretVariable(t *testing.T) {
+	t.Parallel()
+
+	c, ctx := connect(t)
+	defer c.Close()
+
+	ctr := c.Container().From("alpine").
+		WithPlainSecretVariable("SECRET_ENV", "secret-value")
+
+	// Assert env value cannot be read from API
+	v, err := ctr.EnvVariable(ctx, "SECRET_ENV")
+	require.NoError(t, err)
+	require.Equal(t, "", v)
+
+	out, err := ctr.WithExec([]string{"sh", "-c", "echo \"secret=$SECRET_ENV\""}).Stdout(ctx)
+	require.NoError(t, err)
+	require.Equal(t, out, "secret=***\n")
```

Comment:
You can replace that exec with `[]string{"printenv", "SECRET_ENV"}`, which will return `***`, or just `"printenv"` which will contain `SECRET_ENV=***`.

But I prefer testing the value as well to make sure it's all as expected:

```suggestion
	_, err := ctr.WithExec([]string{"sh", "-c", "test $SECRET_ENV = secret-value"}).Sync(ctx)
	require.NoError(t, err)
```